cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -714,7 +715,81 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
                 AbstractVisibilityEditor.prototype.getContext.call(this),
                 {
                     hide_after_due: this.modelVisibility() === 'hide_after_due',
-                    self_paced: this.model.get('self_paced') === true
+                    self_paced: course.get('self_paced') === true
+                }
+            );
+        }
+    });
+
+    ShowCorrectnessEditor = AbstractEditor.extend({
+        templateName: 'show-correctness-editor',
+        className: 'edit-show-correctness',
+        events: {
+            'change input[name=show-correctness]': 'togglePastDueWarning'
+        },
+
+        modelValue: function() {
+            var showCorrectness = this.model.get('show_correctness');
+            if (showCorrectness === 'always' ||
+                showCorrectness === 'never' ||
+                showCorrectness === 'past_due') {
+                return showCorrectness;
+            }
+            return 'always';
+        },
+
+        afterRender: function() {
+            AbstractEditor.prototype.afterRender.call(this);
+            this.setValue(this.modelValue());
+            this.$('input[name=show-correctness]:checked').change();
```

Comment:
It was only there to ensure the `.tip-warning` was shown when the modal is first rendered.  I've removed it with https://github.com/edx/edx-platform/pull/14737/commits/541b3c9d4d39e3d84910ea242c86c9dc4569e069#diff-fecac9c1a5c871b71bafab46e808b363L744.